packages/web3torrent/src/components/form/form-input/FormInput.tsx

```diff
@@ -0,0 +1,86 @@
+import React from "react";
+import { errorStatus, ExtendedFormInputProps, FormInputProps } from "../types";
+import "./FormInput.scss";
+
+class FormInput extends React.Component<FormInputProps, ExtendedFormInputProps> {
+  constructor(props: FormInputProps) {
+    super(props);
+    this.state = {
+      lastChangeEvent: undefined,
+      value: props.value || "",
+      error: props.error,
+      valid: !!(props.error || (props.required && props.value !== undefined))
```

Comment:
`!!` could be replaced by `props.error !== undefined`.

Reply:
I don't remember why I put that like that originally (I copy-pasted the component from the wallet-ui), but until I've got a use case where I can test the change, I don't feel comfortable changing the behaviour of a button that was working okay.

Reply:
It's not a behavioural change. You're coercing to Boolean because you're doing a "truthy" check on props.error.

In any case, it's not a blocker.